File: seg7.py
#!/usr/bin/python3
from gpiozero import LED
from time import sleep
# from random import seed
from random import seed, randint
from datetime import datetime
# seed random number generator
seed(datetime.now())
leda=LED(26)
ledb=LED(12)
ledc=LED(20)
ledd=LED(21)
lede=LED(13)
ledf=LED(19)
ledg=LED(16)
leddp=LED(5)
mylist=[]
thelist=[]
for i in range(5):
	x=randint(0,9)
	x=int(x)
	num=str(x)
	mylist.append(num)
	leda.off()
	ledb.off()
	ledc.off()
	ledd.off()
	lede.off()
	ledf.off()
	ledg.off()
	leddp.off()
	
	if num == "0":
		leda.on()
		ledb.on()
		ledc.on()
		ledd.on()
		lede.on()
		ledf.on()
	if num == "1":
		ledc.on()
		ledb.on()
	if num == "2":
		leda.on()
		ledb.on()
		ledg.on()
		lede.on()
		ledd.on()
	if num == "3":
		leda.on()
		ledb.on()
		ledg.on()
		ledc.on()
		ledd.on()
	if num == "4":
		ledf.on()
		ledg.on()
		ledb.on()
		ledc.on()
	if num == "5":
		leda.on()
		ledf.on()
		ledg.on()
		ledc.on()
		ledd.on()
	if num == "6":
		leda.on()
		ledc.on()
		ledd.on()
		lede.on()
		ledf.on()
		ledg.on()
	if num == "7":
		leda.on()
		ledb.on()
		ledc.on()
	if num == "8":
		leda.on()
		ledb.on()
		ledc.on()
		ledd.on()
		lede.on()
		ledf.on()
		ledg.on()
	if num == "9":
		leda.on()
		ledb.on()
		ledc.on()
		ledf.on()
		ledg.on()
	sleep(0.2)
leda.off()
ledb.off()
ledc.off()
ledd.off()
lede.off()
ledf.off()
ledg.off()
#print(mylist)
answer=input('Enter your answer:')
for i in answer:
	thelist.append(str(i))
#print(thelist)
if (mylist == thelist):
	print('Winner!')
else:
	print('loser!')

File: seg-7.c
#include <stdio.h>
#include <stdlib.h>
#include <wiringPi.h>
#define LEDA 20
#define LEDB 12
#define LEDC 28
#define LEDD 21
#define LEDE 13
#define LEDF 19
#define LEDG 16
#define LEDDP 9

void allOff() {
	digitalWrite(LEDA, LOW);
	digitalWrite(LEDB, LOW);
	digitalWrite(LEDC, LOW);
	digitalWrite(LEDD, LOW);
	digitalWrite(LEDE, LOW);
	digitalWrite(LEDF, LOW);
	digitalWrite(LEDG, LOW);
	digitalWrite(LEDDP, LOW);
}

void initAll() {
	time_t t;
	srand((unsigned) time(&t));
	
	wiringPiSetupGpio ();
	pinMode (LEDA, OUTPUT);
	pinMode (LEDB, OUTPUT);
	pinMode (LEDC, OUTPUT);
	pinMode (LEDD, OUTPUT);
	pinMode (LEDE, OUTPUT);
	pinMode (LEDF, OUTPUT);
	pinMode (LEDG, OUTPUT);
	pinMode (LEDDP, OUTPUT);
}

int main() {
	initAll();
	allOff();
	int number,i,mylist[5],thelist[5];
	for(i=0;i<5;i++) {
		allOff();
		number=rand()%10;
		mylist[i]=number;
		switch (number) {
			case 0: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDC, HIGH);
					digitalWrite(LEDD, HIGH);
					digitalWrite(LEDE, HIGH);
					digitalWrite(LEDF, HIGH);
					break;
			case 1: digitalWrite(LEDC, HIGH);
					digitalWrite(LEDD, HIGH);
					break;
			case 2: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDG, HIGH);
					digitalWrite(LEDE, HIGH);
					digitalWrite(LEDD, HIGH);
					break;
			case 3: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDG, HIGH);
					digitalWrite(LEDC, HIGH);
					digitalWrite(LEDD, HIGH);
					break;
			case 4: digitalWrite(LEDF, HIGH);
					digitalWrite(LEDG, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDC, HIGH);
					break;
			case 5: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDF, HIGH);
					digitalWrite(LEDG, HIGH);
					digitalWrite(LEDC, HIGH);
					digitalWrite(LEDD, HIGH);
					break;
			case 6: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDC, HIGH);
					digitalWrite(LEDD, HIGH);
					digitalWrite(LEDE, HIGH);
					digitalWrite(LEDF, HIGH);
					digitalWrite(LEDG, HIGH);
					break;
			case 7: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDC, HIGH);
					break;
			case 8: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDC, HIGH);
					digitalWrite(LEDD, HIGH);
					digitalWrite(LEDE, HIGH);
					digitalWrite(LEDF, HIGH);
					digitalWrite(LEDG, HIGH);
					break;
			case 9: digitalWrite(LEDA, HIGH);
					digitalWrite(LEDB, HIGH);
					digitalWrite(LEDC, HIGH);
					digitalWrite(LEDF, HIGH);
					digitalWrite(LEDG, HIGH);
					break;
		}
	}
	allOff();
	printf("Enter your answer:\n");
	scanf("%d%d%d%d%d",thelist[0],thelist[1],thelist[2],thelist[3],thelist[4]);
	for(i=0;i<5;i++) {
		if(mylist[i] != thelist[i]) {
			printf("Loser!\n");
			exit(0);
		}
	}
	printf("Winner!\n");
	return 0;
}
